test/fixtures/wpt/dom/events/EventTarget-constructible.any.js

```diff
@@ -23,6 +23,23 @@ test(() => {
   assert_equals(callCount, 2);
 }, "A constructed EventTarget can be used as expected");
 
+test(() => {
+  const target = new EventTarget();
+  const event = new Event("foo");
+
+  function listener(e) {
+    assert_equals(e, event);
+    assert_equals(e.target, target);
+    assert_equals(e.currentTarget, target);
+    assert_array_equals(e.composedPath(), [target]);
+  }
+  target.addEventListener("foo", listener, { once: true });
+  target.dispatchEvent(event);
+  assert_equals(event.target, target);
+  assert_equals(event.currentTarget, null);
+  assert_array_equals(event.composedPath(), []);
+}, "A constructed EventTarget implements dispatch correctly");
```

Comment:
This is the test that checks the behavior. Afaict `event.target` -> not reset after dispatch; `event.currentTarget` -> reset after dispatch.